service/src/repository/user_respository.ts

```diff
@@ -0,0 +1,255 @@
+            },
+            "dob": {
+                S: user?.dob ?? ''
+            },
+            "address": {
+                S: user?.address ?? ''
+            },
+            "imageUrl": {
+                S: user?.imageUrl ?? ''
+            },
+            "createdAt": {
+                S: user.createdAt
+            },
+            "updatedAt": {
+                S: user.updatedAt
+            },
+        },
+        ReturnConsumedCapacity: "TOTAL",
+        TableName: USERS_TABLE,
+    };
+
+    await dbClient.send(new PutItemCommand(params));
+    return getUserByUserId(user.id);
+};
+
+/**
+ * Updates user.
+ *
+ * @param {string} id user id
+ * @param {User} user user properties to be updaed
+ *
+ * @returns updated user.
+ */
+export const updateUser = async (id: string, user: User) : Promise<User> => {
+    const params = {
+        ExpressionAttributeNames: {
+            "#userName": "userName",
+            "#dob": "dob",
+            "#address": "address",
+            "#description": "description",
+            "#imageUrl": "imageUrl",
+            "#updatedAt": "updatedAt",
+        },
+        ExpressionAttributeValues: {
+            ":userName": {
+                S: user.name
+            },
+            ":dob": {
+                S: user.dob ?? ''
+            },
+            ":address": {
+                S: user.address ?? ''
+            },
+            ":description": {
+                S: user.description ?? ''
+            },
+            ":imageUrl": {
+                S: user.imageUrl ?? ''
+            },
+            ":updatedAt": {
+                S: user.updatedAt ?? ''
+            },
+        },
+        Key: {
+            "id": {
+                S: id
+            },
+        },
+        ReturnValues: "ALL_NEW",
+        TableName: USERS_TABLE,
+        UpdateExpression: `SET 
+            #userName = :userName, 
+            #dob = :dob, 
+            #address = :address,
+            #description = :description,
+            #imageUrl = :imageUrl,
+            #updatedAt = :updatedAt`
+    };
+
+    user.updatedAt = new Date().toISOString();
+    await dbClient.send(new UpdateItemCommand(params));
+    return getUserByUserId(id);
+};
+
+/**
+ * Deletes user by id
+ *
+ * @param {string} id user id
+ *
+ * @returns result of delete operation.
+ */
+export const deleteUser = async (id: string) : Promise<{result: string}> => {
+    const params = {
+        Key: {
+            "id": {
+                S: id
+            },
+        },
+        TableName: USERS_TABLE,
+        ReturnValues: 'ALL_OLD'
+    };
+
+    const deleteItemCommandOutput = await dbClient.send(new DeleteItemCommand(params));
+    return deleteItemCommandOutput.Attributes
+        ? { result: `Successfully deleted user: ${id}`}
+        : { result: `User with id: ${id} not found`};
+};
+
+/**
+ * ONLY used in test.
+ * Creates user table.
+ */
+export const createLocalUserTable = async (): Promise<void> => {
```

Comment:
What's your opinion on mixing test-specific code and business flow code? When would you say it's ok to do it, and when is it not ok?